search/includes/classes/class-versioning.php

```diff
@@ -360,6 +364,19 @@ public function add_version( Indexable $indexable ) {
 			return new WP_Error( 'unable-to-get-next-version', 'Unable to determine next index version' );
 		}
 
+		$current_version_count = count( $versions );
+		if ( ( $current_version_count + 1 ) > self::MAX_VERSIONS ) {
```

Comment:
```suggestion
		if ( ( $current_version_count ) >= self::MAX_VERSIONS ) {
```

Writing it this way might be more easibly grokable?